File: ArchitectureComponentsMVVM/app/src/main/java/com/andr/mvvm/repository/PopulateDbAsync.java
package com.andr.mvvm.repository;

import android.os.AsyncTask;

public class PopulateDbAsync   extends AsyncTask<Void, Void, Void> {

    private final UserDao mDao;

    PopulateDbAsync(UserRoomDataBase db) {
        mDao = db.userDao();
    }

    @Override
    protected Void doInBackground(final Void... params) {
        mDao.deleteAll();
        User user = new User("Chandra","SW");
        mDao.insert(user);
        user = new User("Mohan","student");
        mDao.insert(user);
        return null;
    }
}

File: MVVM_Kotlin/app/src/main/java/com/mvvm/kot/Repository/PopulateDBAsync.kt
package com.mvvm.kot.Repository

import android.os.AsyncTask
import com.mvvm.kot.ViewModels.User

    class PopulateDbAsync internal constructor(db: UserRoomDataBase) : AsyncTask<Void, Void, Void>() {

        private val mDao: UserDao

        init {
            mDao = db.userDao()
        }

        override fun doInBackground(vararg params: Void): Void? {
            mDao.deleteAll()
            var user = User("Chandra", "SW")
            mDao.insert(user)
            user = User("Mohan", "student")
            mDao.insert(user)
            return null
        }
    }
